### DISCUSSION TITLE
Lateral and longitudinal friction

### DISCUSSION THREAD

**Discussion by LeXXik** [jrouwe/JoltPhysics.js#88] [Q&A] [answered]
I am trying to understand how to properly fill the curves for vehicle wheels lateral and longitudinal frictions and their meaning:

https://jrouwe.github.io/JoltPhysics/class_wheel_settings_w_v.html#a6d94d1e2a107362a82d5c525d524dafc
![image](https://github.com/jrouwe/JoltPhysics.js/assets/5677782/1ac7fa66-026f-4a92-960a-4c588ed3beb4)

Lateral friction:
The angle between the vehicle direction and the contact velocity. So, if the vehicle is sliding sideways, e.g. drifting to the side, then the direction would be forward and tire side to the side, making the angle to be 90 degrees. Say I want to have full friction if vehicle moves straight, and 0.5, if it slides sideways. Does it mean that at zero degrees the friction should be 1, and at 90 should be 0.5 to make the vehicle drift on turns? What would the curve keys be then? If I assume X is an angle and Y friction, then is this correct for a curve that would have only two keys: [[0, 1], [90, 0.5]]?

LongitudinalFriction:
If I understand it correctly, then the slip ratio is a fraction, meaning a part of the forward velocity that will be lost? Does the curve allow to describe how the friction is changed, based on the forward velocity of the vehicle, e.g. the faster it goes, the less friction it has? If yes, then say at velocity 0.5 I want to linearly start losing friction from 1 to 0.3. What would be the keys for such a curve?

**Comment by jrouwe** [comment, marked as answer]
> If I assume X is an angle and Y friction, then is this correct for a curve that would have only two keys: [[0, 1], [90, 0.5]]?

Correct.

> If I understand it correctly, then the slip ratio is a fraction, meaning a part of the forward velocity that will be lost?

You can see slip ratio as the amount the wheel is spinning relative to the floor. 0 means the wheel has full traction and is rolling perfectly in sync with the ground, 1 is for example when the wheel is locked and sliding over the ground.

> Does the curve allow to describe how the friction is changed, based on the forward velocity of the vehicle, e.g. the faster it goes, the less friction it has?

No it does not. I implemented a common tire model which doesn't look at absolute velocity, but you can easily implement this yourself by calling VehicleConstraint::SetCombineFriction and doing your own calculation. The inTireFriction parameter is what came out of the curves but you can choose to ignore it.

**Reply by jrouwe** [reply]
Ah, I didn't realize this question was for Jolt.JS, the friction callback is on VehicleConstraintCallbacksJS.GetCombinedFriction

**Reply by LeXXik** [reply]
Got it, thank you!

**Reply by kutu** [reply]
In LongitudinalFriction I understood about 0-1 traction, but what the other axis means, and which one is which?

**Reply by jrouwe** [reply]
Longitudinal = in the direction the tire normally rolls in
Lateral = perpendicular to that, i.e. in the direction of the rotation axis of the tire

**Reply by kutu** [reply]
X and Y of LongitudinalFriction, what is what, not Longitudinal / Lateral.
X - is speed of wheel, angular, linear?
Y - 0=full traction / 1=sliding?

**Reply by jrouwe** [reply]
X = slip ratio for long/slip angle for lat (the docs mentioned in the initial post specify what this means)
Y = friction coefficient. 0 is no friction, 1 = full friction (although a tire can go a bit above 1 due to its profile)
